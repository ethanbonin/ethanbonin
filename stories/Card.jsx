import React from "react"
import PropTypes from "prop-types"
import styles from '../styles/card.module.css'
import { Card } from "react-bootstrap"
import Link from 'next/link'

/**
 * Primary UI component for user interaction
 */
export const PrimaryCard = ({ label, href, onClick }) => {
    return (
        <Link href={href}>
            <Card
                style={{ width: "17rem", height: "12rem" }}
                className={styles.card}
                onClick={onClick}
            >
                <Card.Text style={{textAlign: "center"}}>
                    {label.toUpperCase()}
                </Card.Text>
            </Card>
        </Link>
    )
}

PrimaryCard.propTypes = {
    /**
     * Card Title
     */
    label: PropTypes.string.isRequired,
    /**
     * the endpoint that the card will route to
     */
    href: PropTypes.string.isRequired,
}

PrimaryCard.defaultProps = {
    label: "My Awesome Card",
}
