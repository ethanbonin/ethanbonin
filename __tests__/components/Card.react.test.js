// Link.react.test.js
import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import { PrimaryCard } from "@stories/Card"

describe("Card", () => {
    it("Should change colors when hovered", () => {
        const mockCallBack = jest.fn()

        const component = shallow(
            <PrimaryCard label={"My Awesome Card"} href={"/"}/>,
        )
        let tree = toJson(component)
        expect(tree).toMatchSnapshot()

        // manually trigger the callback
        component.simulate("mouseover")
        // re-rendering
        tree = toJson(component)
        expect(tree).toMatchSnapshot()

        // manually trigger the callback
        component.simulate("mouseleave")
        // re-rendering
        tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })
})