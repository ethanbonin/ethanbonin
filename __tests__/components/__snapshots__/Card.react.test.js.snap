// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should change colors when hovered 1`] = `
<Link
  href="/"
>
  <Card
    body={false}
    style={
      Object {
        "height": "12rem",
        "width": "17rem",
      }
    }
  >
    <CardText
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      MY AWESOME CARD
    </CardText>
  </Card>
</Link>
`;

exports[`Card Should change colors when hovered 2`] = `
<Link
  href="/"
>
  <Card
    body={false}
    style={
      Object {
        "height": "12rem",
        "width": "17rem",
      }
    }
  >
    <CardText
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      MY AWESOME CARD
    </CardText>
  </Card>
</Link>
`;

exports[`Card Should change colors when hovered 3`] = `
<Link
  href="/"
>
  <Card
    body={false}
    style={
      Object {
        "height": "12rem",
        "width": "17rem",
      }
    }
  >
    <CardText
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      MY AWESOME CARD
    </CardText>
  </Card>
</Link>
`;
